### DISCUSSION TITLE
Import File Button in CustomTkinter

### DISCUSSION THREAD

**Discussion by lorenzobandini** [TomSchimansky/CustomTkinter#2519] [Q&A] [answered]
hello everyone,
I need to insert a button in my interface to open a dialog and select a file that I then have to work on. Is there any specific button or method to do it?
Thanks

**Comment by dipeshSam** [comment, marked as answer]
Use `filedialog` by Tkinter. Here is sample code:

``` Python
import tkinter as tk
from tkinter import ttk
from tkinter import filedialog as fd
from tkinter.messagebox import showinfo
from customtkinter import  CTk, CTkButton 

# create the root window
root = CTk()
root.title('Tkinter Open File Dialog')
root.resizable(False, False)
root.geometry('300x150')


def select_file():
    filetypes = (
        ('text files', '*.txt'),
        ('All files', '*.*')
    )

    filename = fd.askopenfilename(
        title='Open a file',
        initialdir='/',
        filetypes=filetypes)

    showinfo(
        title='Selected File',
        message=filename
    )


# open button
open_button = CTkButton(
    root,
    text='Open a File',
    command=select_file
)

open_button.pack(expand=True)


# run the application
root.mainloop()

```

**Reply by lorenzobandini** [reply]
Thank you so much, now it's clear to me

**Reply by dipeshSam** [reply]
You are most welcome 😊

Regards.

**Reply by mahanmarwat** [reply]
@dipeshSam Initially in which directory it will look for by specifying this argument? 
initialdir='/'

**Reply by dipeshSam** [reply]
@mahanmarwat Yes. For example, if you pass `initialdir = "C:\\Users\\userName\\Music"`, it will look through folder Music first.

**Reply by mahanmarwat** [reply]
@dipeshSam I am asking about the argument value “/”? Is it current directory or what?

**Reply by dipeshSam** [reply]
This is not the current directory. This is root location. For example, "C/User/UserName/" that's it.
